Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env bash
+#######################################################################
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+#
+# validate-da.sh
+# Description:
+#    Validate the installation and uninstallation of the Dependency Agent.
+#    Validate the enable and disable of the Dependency Agent.
+#######################################################################
+
+set -e
+set -x
+
+# Source utils.sh
+. utils.sh || {
+    echo "ERROR: unable to source utils.sh!"
+    echo "TestAborted" > state.txt
+    exit 0
+}
+
+# Get distro information
+GetDistro
+
+readonly dir_list=(/opt/microsoft/dependency-agent /var/opt/microsoft/dependency-agent /etc/opt/microsoft/dependency-agent /lib/microsoft-dependency-agent /etc/init.d/microsoft-dependency-agent)
+
+readonly da_pid_file="/etc/opt/microsoft/dependency-agent/config/DA_PID"
+readonly uninstaller="/opt/microsoft/dependency-agent/uninstall"
+
+function test_case_cleanup(){
+    [ -f "$uninstaller" ] && eval $uninstaller
+
+    rm -rf "${dir_list[@]}"
+
+    # This can be a NOP if the test exits before calling generate_network_activity
+    kill $network_activity_generator_pid
+
+    LogMsg "Finished cleaning up test cases"
+}
+
+# Calls the cleanup function when interupted or exited
+trap test_case_cleanup SIGINT EXIT SIGHUP SIGQUIT SIGTERM
+
+function fail_test() {
+    LogErr "$*"
+    SetTestStateFailed
+    exit 0
+}
+
+function verify_directory_exists() {
+    failed_assertions=false
+    for dir in "${dir_list[@]}"
+    do
+        if [ -d "$dir" ]; then
+            LogErr "Directory $dir exists."
+            failed_assertions=true
+        fi
+    done
+
+    if $failed_assertions; then
+        LogErr "Some/All directories exists"
+        echo "$failed_assertions"
+    fi
+}
+
+function verify_distro() {
+    case $DISTRO in
+        redhat* | centos* | suse* | debian* | ubuntu*)
+            LogMsg "Supported Distro family: $DISTRO"
+            ;;
+        *)
+            fail_test "Unsupported Distro family: $DISTRO"
```

Comment:
Should this fail, skip, or abort the test suite. A "failed" test implies something is broken, and therefore, requires corrective action.

Reply:
Changed to the condition to skip